### PR TITLE
Baseball

### PR DESCRIPTION
**Pull request by Atlante45**
This PR provides the scripts necessary to run the Baseball demo as well as a list of the assets necessary.

Included are a few other features/bug fix we developed along the way:
- Physics engine is now called at 90fps (instead of 60)
- Fixed a bug in the AssetResourceRequest
- Fixed Vive controllers hand offset
- Fixed Avatar local render option
- Added getJointPosition/getJointRotation for entities
- Added option to ignore the IK for hold actions
- Added option to specify different left/right position/rotation offsets for hold actions

Here's an in-game screenshot of birarda playing with a custom made avatar:
Birarda
Screenshot credit to mister huffman

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4e43c4e0b0e8d66b3ea56a0ce18917bfe61f72e available at: https://gist.github.com/f3519d991e5b33239d30<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6562.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6562.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1d498af76e673e74efca70bbce4a617b8d656061 available at: https://gist.github.com/f3519d991e5b33239d30<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6562.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6562.dmg

File: examples/baseball/assets.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      "atp_url": "atp:830312930577cb1ea36ba2d743e957debbacceb441b20addead5a6faa05a3771.wav"
+    },
+    "pop4.wav": {
+      "s3_url": "http://hifi-public.s3.amazonaws.com/birarda/baseball/fireworks/pop4.wav",
+      "atp_url": "atp:62e80d0a9f084cf731bcc66ca6e9020ee88587417071a281eee3167307b53560.wav"
+    },
+    "fire1.wav": {
+      "s3_url": "http://hifi-public.s3.amazonaws.com/birarda/baseball/fireworks/fire1.wav",
+      "atp_url": "atp:ee6afe565576c4546c6d6cd89c1af532484c9b60ab30574d6b40c2df022f7260.wav",
+      "attribution": "CC BY 3.0 - Credirt: dcsimon - https://www.freesound.org/people/dcsimon/sounds/160720/"
```
Comment (birarda) [inline examples/baseball/assets.json]: @Atlante45 this is my fault but can you fix `Credirt` here?

Comment (Atlante45) [inline examples/baseball/assets.json]: That's not how you spell it?

Comment (huffman) [inline examples/baseball/assets.json]: No, it's Kredit.

**Comment by AndrewMeadows** [comment]
I skipped the JS code but looked at the C++ changes: I don't see any problems.  In particular regarding the change to 1/90th sec substeps: if it takes 1 msec for the physics simulation to compute one substep at 60Hz then it will tend to take 1 msec to execute the substep at 90Hz -- for most cases the duration of the step does NOT influence the cost of the substep (*), so we'll spend 50% more total time simulating physics per second at 90Hz than at 60Hz -- not a problem as long as our simulations continue to be simple.

(*) The exception to this rule is a simulation that has many fast moving objects that penetrate into other objects -- stepping at 90Hz will tend to reduce the penetration depths since more contacts will collide and bounce correctly before the objects actually overlap.  It is computationally cheaper to resolve non-overlapping (**) collisions than it is to resolve penetrations.  For such a case the cost of a single 1/90th sec substep will tend to be cheaper than one 1/60th sec substep.

(**) Where non-overlapping means the object shapes overlap no further than their "collision margin".  The cost of the GJK collision detection algorithm is different for the two cases.

**Comment by imgntn** [comment]
is there a stadium domain or SVO?

**Comment by birarda** [comment]
@imgntn There's a domain that will hold all of this `sports.highfidelity.io` but it hasn't been completely set up yet because we needed this merged first.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84bc947854fe4d998a2b06c0bdf35bb250d03077 available at: https://gist.github.com/f3519d991e5b33239d30<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6562.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6562.dmg

**Pull request merged**